db_stress_tool/db_stress_test_base.cc

```diff
@@ -2541,7 +2542,13 @@ void StressTest::Open(SharedState* shared) {
         fflush(stderr);
       }
       assert(s.ok());
-      db_ = txn_db_;
+
+      // Do not swap the order of the following.
+      {
+        db_ = txn_db_;
+        db_aptr_.store(txn_db_, std::memory_order_release);
```

Comment:
Just wanted to understand, how does this fixes the race condition?

Reply:
This ensures that, when `MultiOpsTxnsStressTest::VerifyPkSkFast()` is invoked in bg compaction thread for a compaction that occurs when the db is being opened, the check in the above said method will correctly detect if `db_` is nullptr or not.

Reply:
Ah ok got it. And this will avoid TSAN error as well. Thanks for the clarification.